lib/eclipse/EclipseState/Schedule/TimeMap.cpp

```diff
@@ -189,57 +239,71 @@ namespace Opm {
         }
     }
 
-    void TimeMap::addFromDATESKeyword( const DeckKeyword& DATESKeyword ) {
+    std::time_t TimeMap::timeTFromEclipse(const DeckRecord &dateRecord) {
+        static const std::string errorMsg("The datarecord must consist of the four values "
```

Comment:
This will be validated by the `getItem( )` callls anyway. As I see it this function can be significantly simplifed:

```C++
const auto &dayItem = dateRecord.getItem(0);
const auto &monthItem = dateRecord.getItem(1);
const auto &yearItem = dateRecord.getItem(2);
const auto &timeItem = dateRecord.getItem(3);

std::time_t date= mkdate(yearItem.get<int>(0), 
                                          eclipseMonthIndices().at(monthItem.get<std::string>(0)) + 1, 
                                          dayItem.get<int>(0));
return date;
```